apps/examples/wifi_manager/wm_test/wm_test_tls_client.c

```diff
@@ -0,0 +1,266 @@
+		WT_LOG(TAG, " failed ! mbedtls_net_connect returned %d", ret);
+		goto exit;
+	}
+	WT_LOG(TAG, " ok");
+
+	/* 4. Setup stuff */
+	WT_LOG(TAG, " Setting up the SSL/TLS structure...");
+	fflush(stdout);
+
+	if ((ret = mbedtls_ssl_config_defaults(&conf, MBEDTLS_SSL_IS_CLIENT,
+	  MBEDTLS_SSL_TRANSPORT_STREAM, MBEDTLS_SSL_PRESET_DEFAULT)) != 0) {
+		WT_LOG(TAG, " failed ! mbedtls_ssl_config_defaults returned %d", ret);
+		goto exit;
+	}
+	WT_LOG(TAG, " ok");
+
+	mbedtls_ssl_conf_authmode(&conf, MBEDTLS_SSL_VERIFY_REQUIRED);
+	mbedtls_ssl_conf_ca_chain(&conf, &cacert, NULL);
+	mbedtls_ssl_conf_rng(&conf, mbedtls_ctr_drbg_random, &ctr_drbg);
+	mbedtls_ssl_conf_dbg(&conf, my_debug, stdout);
+
+	if ((ret = mbedtls_ssl_setup(&ssl, &conf)) != 0) {
+		WT_LOG(TAG, " failed ! mbedtls_ssl_setup returned %d", ret);
+		goto exit;
+	}
+
+	mbedtls_ssl_set_bio(&ssl, &server_fd, mbedtls_net_send, mbedtls_net_recv, mbedtls_net_recv_timeout);
+
+	/*5. Handshake */
+	WT_LOG(TAG, " Performing the SSL/TLS handshake...");
+	fflush(stdout);
+
+	while ((ret = mbedtls_ssl_handshake(&ssl)) != 0) {
+		if (ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
+			WT_LOG(TAG, " failed ! mbedtls_ssl_handshake returned -0x%x", (unsigned int)-ret);
+			goto exit;
+		}
+	}
+	WT_LOG(TAG, " ok");
+
+	/* 6. Verify the server certificate */
+	WT_LOG(TAG, "	. Verifying peer X.509 certificate...");
+	if ((flags = mbedtls_ssl_get_verify_result(&ssl)) != 0) {
+		char vrfy_buf[512];
+		WT_LOG(TAG, " failed\n");
+		mbedtls_x509_crt_verify_info(vrfy_buf, sizeof(vrfy_buf), " ! ", flags);
+		WT_LOG(TAG, "%s", vrfy_buf);
+	} else
+		WT_LOG(TAG, " ok");
+
+	/* 7. Write the GET request */
+send_request:
+	WT_LOG(TAG, "	> Write to server:");
+	fflush(stdout);
+
+	int remain = msg_size;
+	int cnt = 1;
+	int total_sent = 0; 
+	WT_LOG(TAG, "_tls_send_data start");
+	while (remain > 0) {
+		int send_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		do {
+			ret = mbedtls_ssl_write(&ssl, buf, send_size);
+		} while (ret == MBEDTLS_ERR_SSL_WANT_READ || ret == MBEDTLS_ERR_SSL_WANT_WRITE);
+
+		if (ret < 0) {
+			WT_LOG(TAG, " failed  ! mbedtls_ssl_write returned %d", ret);
+			goto exit;
+		}
+		total_sent += ret;
+		remain -= ret;
+		WT_LOG(TAG, "Packet Number: %d Sent: %d Total Sent: %d Remaining: %d", cnt, ret, total_sent, remain);
+		cnt++;
+	}
+	WT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+
+	/* 7. Read the HTTP response */
+	WT_LOG(TAG, "   < Read from server:");
+	fflush(stdout);
+
+	remain = msg_size;
+	cnt = 1;
+	int total_received = 0;
+	WT_LOG(TAG, "_tls_recv_data start");
+	while (remain > 0) {
+		do {
+			int read_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+			ret = mbedtls_ssl_read(&ssl, buf, read_size);
+		} while (ret == MBEDTLS_ERR_SSL_WANT_READ || ret == MBEDTLS_ERR_SSL_WANT_WRITE);
+
+		if (ret <= 0) {
+			switch (ret) {
+			case MBEDTLS_ERR_SSL_TIMEOUT:
+				WT_LOG(TAG, " timeout");
+				if (retry_left-- > 0) {
+					goto send_request;
```

Comment:
this may make an infinite loop b/w recv timeout and send success

Reply:
Changes done as suggested